common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java

```diff
@@ -65,38 +65,26 @@ public static ThreadFactory createThreadFactory(String threadPoolPrefix) {
   public static EventLoopGroup createEventLoop(IOMode mode, int numThreads, String threadPrefix) {
     ThreadFactory threadFactory = createThreadFactory(threadPrefix);
 
-    switch (mode) {
-      case NIO:
-        return new NioEventLoopGroup(numThreads, threadFactory);
-      case EPOLL:
-        return new EpollEventLoopGroup(numThreads, threadFactory);
-      default:
-        throw new IllegalArgumentException("Unknown io mode: " + mode);
-    }
+    return switch (mode) {
+      case NIO -> new NioEventLoopGroup(numThreads, threadFactory);
+      case EPOLL -> new EpollEventLoopGroup(numThreads, threadFactory);
```

Comment:
It seems not treat the default path.

Reply:
<img width="854" alt="image" src="https://github.com/apache/spark/assets/1475305/fc3510f8-30fd-4674-949c-f10de8e8ba1d">

Reply:
Additionally, if IOMode is extended and this method is not updated in sync, a compile error will occur to remind the developer to add the corresponding case branch, so, for this enum matching scenario, `default` is not necessary:)

Reply:
Yeah. Thank you for the explanation. The origin behavior will throw exception. What's now?

Reply:
Compilation failed if we just add a new IOMode item, like 

```
[error] /Users/yangjie01/SourceCode/git/spark-mine-sbt/common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java:68:1:  错误: switch 表达式不包含所有可能的输入值
[error]     return switch (mode) {
[error]            ^
[warn] javac exited with exit code 1
[error] (network-common / Compile / compileIncremental) javac returned non-zero exit code
[error] Total time: 7 s, completed 2023年10月13日 下午3:42:54

```